mesh/ncmesh.cpp

```diff
@@ -4435,6 +4435,11 @@ void CoarseFineTransformations::GetCoarseToFineMap(
       coarse_to_fine.GetJ()[i] = cf_j[i].two;
    }
 
+   if (fine_mesh.GetLastOperation() == Mesh::Operation::DEREFINE)
```

Comment:
This seems like a hacky solution. Unfortunately I'm not the author of `CoarseFineTransformations::GetCoarseToFineMap` so I'm not sure what better solution to suggest (the method is not documented and I'm having a hard time understanding it). Maybe split the method into several methods and avoid calling those that don't make sense for derefinement? Maybe @v-dobrev or @pazner can advise here?